pkg/controller/storagecluster/storagecluster.go

```diff
@@ -631,14 +631,16 @@ func (c *Controller) syncStorageCluster(
 	}
 
 	pxVer30, _ := version.NewVersion("3.0")
-	if pxutil.GetPortworxVersion(cluster).GreaterThanOrEqual(pxVer30) { // Preflight should only run on PX version 3.0.0 and above
+	if pxutil.IsFreshInstall(cluster) && pxutil.GetPortworxVersion(cluster).GreaterThanOrEqual(pxVer30) { // Preflight should only run on PX version 3.0.0 and above
 		// If preflight failed, or previous check failed, reconcile would stop here until issues got resolved
 		if err := c.runPreflightCheck(cluster); err != nil {
 			if updateErr := c.updateStorageClusterState(cluster, corev1.ClusterStateDegraded); updateErr != nil {
 				logrus.Errorf("failed to update StorageCluster status. %v", updateErr)
 			}
 			return fmt.Errorf("preflight check failed for StorageCluster %v/%v: %v", cluster.Namespace, cluster.Name, err)
 		}
+	} else {
+		cluster.Annotations[pxutil.AnnotationPreflightCheck] = "false"
```

Comment:
why is the explicit setting to false needed now and not before?

Reply:
Didn't have it before ... With the new operator we set this to true cause it does not exist.  If we skip pre-flight it never gets set to false and will stay true.   I am explicitly setting it for consistency.  We will skip pre-flight anyways but wanted to make sure the annotation was set.

Reply:
@jrivera-px let's set it to false only if it is not set or value is empty.